inc/blocks/ab-test/register.php

```diff
@@ -113,6 +113,32 @@ function render_block( array $attributes, ?string $inner_content = '' ) : string
 		$winner = $results['winner'] ?? false;
 	}
 
+	if ( is_preview() || isset( $_GET['set_test'] ) ) {
```

Comment:
I think this means I could use the `set_test` query string parameter even if I'm not logged in or able to edit the content, would be good to check permissions here before showing the preview mode.

Reply:
I have added a check for if a user can edit posts and wrapped the preview check in the if statement. PR has been updated